bn-apps/memberships-management/membership-service-contracts-and-states/src/main/kotlin/net/corda/businessnetworks/membership/states/Membership.kt

```diff
@@ -32,74 +46,70 @@ class MembershipContract : Contract {
 
     override fun verify(tx : LedgerTransaction) {
         val command = tx.commands.requireSingleCommand<Commands>()
-        val output = tx.outputs.single()
-        val outputState = output.data as MembershipState<*>
+        val outputMembership = tx.outputsOfType<MembershipState<*>>().single()
 
         requireThat {
-            "Modified date has to be greater or equal to the issued date" using (outputState.modified >= outputState.issued)
-            "Both BNO and member have to be participants" using (outputState.participants.toSet() == setOf(outputState.member, outputState.bno))
-            "Output state has to be validated with $CONTRACT_NAME" using (output.contract == CONTRACT_NAME)
```

Comment:
Isn't this line checking the output contract useful? Without it I can amend the state, give the output a different contract, no one's gonna notice and then run away with it?

Reply:
Good spot. Let me add it back